packages/kit/src/core/sync/write_types/index.js

```diff
@@ -274,33 +279,52 @@ function process_node(node, outdir, params) {
 		}
 
 		server_data = get_data_type(node.server, 'null', proxy);
-		server_load = `Kit.ServerLoad<${params}, ${get_parent_type(
-			node,
-			'LayoutServerData'
-		)}, OutputData>`;
 
-		if (proxy) {
-			const types = [];
-			for (const method of ['POST', 'PUT', 'PATCH']) {
-				if (proxy.exports.includes(method)) {
-					// If the file wasn't tweaked, we can use the return type of the original file.
-					// The advantage is that type updates are reflected without saving.
-					const from = proxy.modified
-						? `./proxy${replace_ext_with_js(basename)}`
-						: path_to_original(outdir, node.server);
-
-					types.push(`Kit.AwaitedErrors<typeof import('${from}').${method}>`);
+		const parent_type = `${prefix}ServerParentData`;
+
+		declarations.push(
+			`type ${parent_type} = EnsureParentData<${get_parent_type(node, 'LayoutServerData')}>;`
+		);
+
+		// +page.js load present -> server can return all-optional data
+		const output_data_shape =
+			node.shared || (!is_page && all_pages_have_load)
+				? `Partial<App.PageData> & Record<string, any> | void`
+				: `OutputDataShape<${parent_type}>`;
+		exports.push(
+			`export type ${prefix}ServerLoad<OutputData extends ${output_data_shape} = ${output_data_shape}> = Kit.ServerLoad<${params}, ${parent_type}, OutputData>;`
+		);
+
+		exports.push(`export type ${prefix}ServerLoadEvent = Parameters<${prefix}ServerLoad>[0];`);
+
+		if (is_page) {
+			let errors = 'unknown';
+			if (proxy) {
+				const types = [];
+				for (const method of ['POST', 'PUT', 'PATCH']) {
```

Comment:
why isn't `DELETE` included here?

Reply:
(i'm sure there's a reason, i just mention it because it wasn't clear — maybe a comment would help)